File: src/main/java/com/prog5/service/ReservationService.java
package com.prog5.service;

import com.prog5.Item;
import com.prog5.Reservation;

import java.time.LocalDate;
import java.util.List;

public record ReservationService(List<Item> items) {

    public boolean reserveItem(String itemName, LocalDate startDate, LocalDate endDate) {
        for (Item item : items) {
            if (item.getName().equals(itemName)) {
                return Reservation.reserveItem(item, startDate, endDate);
            }
        }
        return false;
    }
}

File: ReservationService.py
from datetime import date
from typing import List
from model import Item, Reservation


class ReservationService:
    def __init__(self, items: List[Item]):
        self.items = items

    def reserve_item(self, item_name: str, start_date: date, end_date: date) -> bool:
        for item in self.items:
            if item.name == item_name:
                if Reservation.can_reserve(item, start_date, end_date):
                    reservation = Reservation(item, start_date, end_date)
                    item.add_reservation(reservation)
                    return True
        return False
